### PR TITLE
Do not append zero-valued reputation updates to the log

### LINKED ISSUE #354: 0 value reputation update entries may be obsolete
In the reputation update log can we _not_ log `0` value updates?

### PR DESCRIPTION
**Pull request by kronosapiens**
Closes #354 

Implemented at the `ColonyNetwork` level rather than the `ColonyTask` level for generality.

### CODE REVIEW

File: contracts/ColonyNetwork.sol [hunk 1]
```diff
@@ -267,6 +267,10 @@ contract ColonyNetwork is ColonyNetworkStorage {
   calledByColony
   skillExists(_skillId)
   {
+    if (_amount == 0) {
+      return;
+    }
```
Comment (elenadimitrova) [inline contracts/ColonyNetwork.sol]: That helps with consistency however I think we're still paying a rather large gas price for the external contract call to here. Which was one of the main benefits of this work. Could you at least test this implementation in comparison with logic sitting in `ColonyTask` contract for `finalizeTask` call? That should help us outweigh cost vs complexity.

Comment (kronosapiens) [reply contracts/ColonyNetwork.sol]: Sure!

Comment (elenadimitrova) [reply contracts/ColonyNetwork.sol]: We've merged a fix for the gas reporter so if you rebase you should have the gasCost tests reporting all calls properly now.

Comment (elenadimitrova) [reply contracts/ColonyNetwork.sol]: Okay, let's proceed with keeping the logic in the Network contract

Comment (area) [reply contracts/ColonyNetwork.sol]: > Noting that putting the skip in ColonyFunding means that other sources of reputation updates (for example, miner rewards) currently will not be skipped if zero-valued.

As-is, mining rewards are not skipped, even if they're zero. However, I think that's a good thing, as I have definitely assumed in reputation disputes that there is always at least one entry in the update log...

Comment (kronosapiens) [reply contracts/ColonyNetwork.sol]: Hrm, but all the tests passed... (the skip is currently at the network level, i.e. affects all updates).

Comment (elenadimitrova) [reply contracts/ColonyNetwork.sol]: Perhaps we don't have a test that completes a ReputationMiningCycle and confirms a hash with an empty log?

**Pull request merged**